### PR TITLE
[benchmark] CheckResults with auto-generated error message

### PR DESCRIPTION
**Pull request by palimondo**
In order to minimize impact of results checking on test performance, this removes the @autoclosure for error message.

Added new version of `CheckResults` that takes only `resultsMatch: Bool` - rest of the parameters are defaulted to `StaticString`s for method and file name, plus line number. Old method was deprecated, but left in place as tool for debugging failing checks. All tests were move to use the new method.

Design: I assume that `StaticString` is passed to `CheckResults` as a direct reference, without any retain/release overhead that would impact the measurements. The method is also marked to `@inline(__always)`.

This is a followup to @dabrahams' experiment in #9298. 

Please review. cc @eeckstein, @slavapestov, @gottesmm

### CODE REVIEW

**Comment by dabrahams** [comment]
@swift-ci Please smoke benchmark

**Comment by dabrahams** [comment]
> Design: I assume that StaticString is passed to CheckResults as a direct reference, without any retain/release overhead that would impact the measurements. The method is also marked to @inline(__always).

@palimondo: Please don't assume anything; instead run the benchmarks locally with and without your calls to checkResults and ensure that they are not having any significant impact on timing.  Note also that this needs to be the case for -Onone as well as -O. I don't personally care about -Ounchecked performance, though others might, but if the -O results are acceptable -Ounchecked is very unlikely to be different.

**Comment by palimondo** [comment]
@dabrahams I’m assuming in a sense that that’s what makes theoretical sense to me without being familiar with the details of the StaticString implementation. I was hoping someone with knowledge about that will chime in…
BTW It takes over 4 hours for me to get a build on my Late 2008 MacBook Pro.

**Comment by palimondo** [comment]
@dabrahams On a quick glance these benchmark results look identical to those you got on #9298 when considering the `Sequence` benchmarks (regressions and improvements). 

But I don't seem to get as many in -10% to -5% Delta, but when I look at the reported times, they look do be the same.

**Comment by palimondo** [comment]
I think I'll pack it for today. Its 11:30PM here. Good night!

**Comment by dabrahams** [comment]
@swift-ci Please benchmark

**Comment by dabrahams** [comment]
@swift-ci Please benchmark

**Comment by dabrahams** [comment]
Clearly `StringHasSuffix` is not doing very much work; this change seems to interfere with its ability to measure anything.  @eeckstein any thoughts.

**Comment by eeckstein** [comment]
Yeah, I think we can ignore this for now. Maybe we can make this benchmark more reliable by making the test strings larger.

**Comment by dabrahams** [comment]
@eeckstein Because I'm not sure what you mean by “ignore this” I'm going to cop out and leave it up to you whether to merge this PR or not 😉

**Comment by palimondo** [comment]
@eeckstein Please merge?

File: benchmark/utils/TestsUtils.swift [hunk 1]
```diff
@@ -49,11 +49,31 @@ public func Random() -> Int64 {
...
 }
 
-public func CheckResults(_ resultsMatch: Bool, _ message: @autoclosure () -> String) {
-  guard resultsMatch else {
-    print(message())
-    abort()
-  }
+@inline(__always)
+public func CheckResults(
+    _ resultsMatch: Bool,
```
Comment (xwu) [inline benchmark/utils/TestsUtils.swift]: Nit: indenting in this file is two spaces.

File: benchmark/utils/TestsUtils.swift [hunk 2]
```diff
@@ -49,11 +49,31 @@ public func Random() -> Int64 {
...
+    file: StaticString = #file,
+    function: StaticString = #function,
+    line: Int = #line
+    ) {
```
Comment (xwu) [inline benchmark/utils/TestsUtils.swift]: Nit: closing paren here should be de-indented.

File: benchmark/utils/TestsUtils.swift [hunk 3]
```diff
@@ -49,11 +49,31 @@ public func Random() -> Int64 {
...
+}
+
+// Due to potential overhead of passing closures around on the
+// performance measurements, this version is now deprecated
+@available(*, deprecated,
+  message: "For debugging test failures only! Use the version without message.")
+public func CheckResults(
+    _ resultsMatch: Bool,
+    _ message: @autoclosure () -> String
+    ) {
```
Comment (xwu) [inline benchmark/utils/TestsUtils.swift]: Also here.

**Comment by eeckstein** [comment]
I think it's fine to merge

**Comment by palimondo** [comment]
@eeckstein I’ve noticed the magic spell used by others is @swift-ci Please smoke test and merge

**Comment by eeckstein** [comment]
@swift-ci Please smoke test and merge

**Comment by eeckstein** [comment]
@swift-ci Please smoke test and merge

**Comment by palimondo** [comment]
Eric, Dave, thank you!

**Pull request merged**